lib/config/wrapper.go

```diff
@@ -183,11 +196,29 @@ func (w *wrapper) RawCopy() Configuration {
 
 // Replace swaps the current configuration object for the given one.
 func (w *wrapper) Replace(cfg Configuration) (Waiter, error) {
+	w.initChange()
 	w.mut.Lock()
 	defer w.mut.Unlock()
 	return w.replaceLocked(cfg.Copy())
 }
 
+func (w *wrapper) initChange() {
+	w.replaceCond.L.Lock()
+
+	if w.replaceCounter == math.MaxUint32 {
+		w.replaceCounter = w.replaceCounter - w.currentCounter
+		w.currentCounter = 0
+	}
+	myCounter := w.replaceCounter
+	w.replaceCounter++
+
+	for myCounter > w.currentCounter {
+		w.replaceCond.Wait()
+	}
+
+	w.replaceCond.L.Unlock()
```

Comment:
This whole thing is non-obvious in what it accomplishes and could use a couple of lines of comments about what & why ("ensures config changes are applied in order because ..."). Also I'd use int64 and not require any logic about it wrapper (borderline non-required even in the int32 case). Oh, and assuming I understood the sequencing purpose correctly, the `initChange` name is kinda nondescript. `awaitQueuedOrder` or something?